### PR TITLE
feat: capitalize first letter of each bullet

### LINKED ISSUE #360: Capitalize first letter of each bullet
Much easier to forget to do manually with conventional commits.

### PR DESCRIPTION
**Pull request by maelle**
Fix #360

### CODE REVIEW

File: R/update-news.R [hunk 1]
```diff
@@ -354,6 +355,35 @@ default_type <- function() {
   "Uncategorized"
 }
 
+capitalize_description <- function(df) {
+  df$description
+
+  # leave code alone
+  if (substr(df$description, 1, 1) == "`") {
```
Comment (krlmlr) [inline R/update-news.R:362]: `grepl("^`")` ?

File: R/update-news.R [hunk 2]
```diff
@@ -354,6 +355,35 @@ default_type <- function() {
   "Uncategorized"
 }
 
+capitalize_description <- function(df) {
+  df$description
```
Comment (krlmlr) [inline R/update-news.R:359]: Dead code?

File: R/update-news.R [hunk 3]
```diff
@@ -354,6 +355,35 @@ default_type <- function() {
...
+
+  # leave code alone
+  if (substr(df$description, 1, 1) == "`") {
+    return(df)
+  }
+
+  # leave package name alone
+  pkg_name <- desc::desc_get("Package")
+  start_with_pkg <- (grepl(sprintf("^%s ", pkg_name), df$description) ||
+    grepl(sprintf("^%s's", pkg_name), df$description))
```
Comment (krlmlr) [inline R/update-news.R:369]: `"^%s(?: |'s)"` ? Extract regex to separate variable? And perhaps combine with regex above?

File: R/update-news.R [hunk 4]
```diff
@@ -354,6 +355,35 @@ default_type <- function() {
...
+    toupper(substr(df$description, 1, 1)),
+    substr(df$description, 2, nchar(df$description))
+  )
+  df
+}
+
+capitalize_news <- function(news_items) {
+  news_item <- split(news_items, seq_len(nrow(news_items))) %>%
+    purrr::map(capitalize_description)
+  do.call(rbind, news_item)
```
Comment (krlmlr) [inline R/update-news.R:384]: I don't understand the need to split here.

`map_dfr()` ?

File: R/update-news.R [hunk 5]
```diff
@@ -354,6 +355,35 @@ default_type <- function() {
...
+  start_with_pkg <- (grepl(sprintf("^%s ", pkg_name), df$description) ||
+    grepl(sprintf("^%s's", pkg_name), df$description))
+  if (start_with_pkg) {
+    return(df)
+  }
+
+  df$description <- paste0(
+    toupper(substr(df$description, 1, 1)),
+    substr(df$description, 2, nchar(df$description))
+  )
```
Comment (krlmlr) [inline R/update-news.R:377]: Surprisingly tricky. I was thinking about a rematch2 solution, which could also include the `grepl()` logic from above, and allow for vectorization (and more compact code) perhaps?

"Bury the logic in one complex regex..." 🤔

Comment (maelle) [reply R/update-news.R:377]: I'm re-reading `?toupper` and actually one of the examples there is very similar to what I did above (which does not mean there are no better solutions!).

Comment (maelle) [reply R/update-news.R:377]: One simple solution would be to add a dependency on https://cran.r-project.org/web/packages/snakecase/index.html but that might be overkill for one call only.

File: R/update-news.R (hunk 4 above)
Comment (maelle) [reply R/update-news.R:384]: `map_dfr()` requires dplyr which fledge does not depend upon hence the awkward split.
Should I define a helper `map_dfr` that would hide the split?

Comment (maelle) [reply R/update-news.R]: (I added such a helper)

File: R/update-news.R (hunk 3 above)
Comment (maelle) [reply R/update-news.R]: In the end I realized the "code" regex was useless as `toupper()` will not touch those anyway.

**Comment by maelle** [comment]
Are you still in favor of replacing this with rematch2 or is it fine as is?

**Comment by krlmlr** [comment]
Thanks! No need to dive deeper here.

**Pull request merged**